checker/src/org/checkerframework/checker/index/upperbound/UpperBoundAnnotatedTypeFactory.java

```diff
@@ -84,14 +84,14 @@ public UpperBoundAnnotatedTypeFactory(BaseTypeChecker checker) {
         BOTTOM = AnnotationBuilder.fromClass(elements, UpperBoundBottom.class);
         POLY = AnnotationBuilder.fromClass(elements, PolyUpperBound.class);
 
-        addAliasedAnnotation(IndexFor.class, createLTLengthOfAnnotation());
-        addAliasedAnnotation(IndexOrLow.class, createLTLengthOfAnnotation());
-        addAliasedAnnotation(IndexOrHigh.class, createLTEqLengthOfAnnotation());
-        addAliasedAnnotation(SearchIndexFor.class, createLTLengthOfAnnotation());
-        addAliasedAnnotation(NegativeIndexFor.class, createLTLengthOfAnnotation());
-        addAliasedAnnotation(LengthOf.class, createLTEqLengthOfAnnotation());
-        addAliasedAnnotation(PolyAll.class, POLY);
-        addAliasedAnnotation(PolyIndex.class, POLY);
+        addAliasedAnnotation(IndexFor.class, createLTLengthOfAnnotation(), true);
+        addAliasedAnnotation(IndexOrLow.class, createLTLengthOfAnnotation(), true);
+        addAliasedAnnotation(IndexOrHigh.class, createLTEqLengthOfAnnotation(), true);
+        addAliasedAnnotation(SearchIndexFor.class, createLTLengthOfAnnotation(), true);
+        addAliasedAnnotation(NegativeIndexFor.class, createLTLengthOfAnnotation(), true);
+        addAliasedAnnotation(LengthOf.class, createLTEqLengthOfAnnotation(), true);
+        addAliasedAnnotation(PolyAll.class, POLY, true);
```

Comment:
Why do polymorphic qualifiers need to copy values? If they do, this change should also happen in every other checker that uses polymorphic qualifiers (almost all checkers). If not, then it shouldn't happen here.

Reply:
My bad. Will remove in next iteration. It doesn't do any harm here thought, since there is no annotation arguments to copy.